tfe/resource_tfe_agent_pool_allowed_workspaces.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package tfe
+
+import (
+	"fmt"
+	"github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"log"
+)
+
+func resourceTFEAgentPoolAllowedWorkspaces() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFEAgentPoolAllowedWorkspacesCreate,
+		Read:   resourceTFEAgentPoolAllowedWorkspacesRead,
+		Update: resourceTFEAgentPoolAllowedWorkspacesUpdate,
+		Delete: resourceTFEAgentPoolAllowedWorkspacesDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"agent_pool_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"allowed_workspace_ids": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceTFEAgentPoolAllowedWorkspacesCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(ConfiguredClient)
+
+	apID := d.Get("agent_pool_id").(string)
+
+	// Create a new options struct.
+	options := tfe.AgentPoolUpdateAllowedWorkspacesOptions{}
```

Comment:
```suggestion
	options := tfe.AgentPoolAllowedWorkspacesUpdateOptions{}
```

This and other lint errors should be resolved by https://github.com/hashicorp/go-tfe/pull/701